v2/client/http_receiver_test.go

```diff
@@ -0,0 +1,67 @@
+package client_test
+
+import (
+	"context"
+	"errors"
+	"net/http"
+	"net/http/httptest"
+	"testing"
+
+	cloudevents "github.com/cloudevents/sdk-go/v2"
+	"github.com/cloudevents/sdk-go/v2/client"
+)
+
+func TestEventReceiverServeHTTP_WithContext(t *testing.T) {
+	type ctxKey string
+	const ctxKeyTest ctxKey = "testKey"
+
+	middleware := func(next http.Handler) http.Handler {
+		return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+			ctx := r.Context()
+			ctx = context.WithValue(ctx, ctxKeyTest, "testValue")
+			next.ServeHTTP(w, r.WithContext(ctx))
+		})
+	}
+
+	handler := func(ctx context.Context) error {
+		v, ok := ctx.Value(ctxKeyTest).(string)
+		if !ok {
+			t.Errorf("invalid context value type: %v", v)
+			return errors.New("invalid context")
+		}
+		if v != "testValue" {
+			t.Errorf("invalid context value: %s", v)
+			return errors.New("invalid context")
+		}
+		return nil
+	}
+
+	p, err := cloudevents.NewHTTP()
+	if err != nil {
+		t.Fatal(err)
+	}
+	rh, err := client.NewHTTPReceiveHandler(context.Background(), p, handler)
```

Comment:
and rename this with httpHandler

Reply:
I thought `eventReceiver` was appropriate.
`NewHTTPReceiveHandler` returns `*EventReceiver`
How about that.

Reply:
Oh, sorry I was wrong, conflicts with above.